pkg/pipelines/config/validate.go

```diff
@@ -139,6 +143,10 @@ func (vv *validateVisitor) Service(app *Application, env *Environment, svc *Serv
 	if err := validateName(svc.Name, svcPath); err != nil {
 		vv.errs = append(vv.errs, err)
 	}
+
+	if len(svc.Name) > 40 {
```

Comment:
This should be 47 I think.

Also, it might be nice to extract it to a nicely named constant?